### PR TITLE
Clean up EmbulkEmbed and EmbulkService

### PR DESCRIPTION
**Pull request by dmikurube**
This is just refactoring. Can you have a look, @sakama @kamatama41 ?

* Move `EmbulkService.standardModuleList` to `EmbulkEmbed`
* Reduce usage of Guava
* Use Java 8 `java.util.function.Function` instead of Guava `Function`

----

This will use https://github.com/embulk/guice-bootstrap/pull/12.

### CODE REVIEW

File: embulk-core/src/main/java/org/embulk/EmbulkEmbed.java [hunk 1]
```diff
@@ -1,165 +1,240 @@
...
 import org.embulk.exec.ExecutionResult;
+import org.embulk.exec.ExtensionServiceLoaderModule;
 import org.embulk.exec.GuessExecutor;
 import org.embulk.exec.PartialExecutionException;
 import org.embulk.exec.PreviewExecutor;
 import org.embulk.exec.PreviewResult;
 import org.embulk.exec.ResumeState;
+import org.embulk.exec.SystemConfigModule;
 import org.embulk.exec.TransactionStage;
 import org.embulk.guice.Bootstrap;
```
Comment (kamatama41) [inline embulk-core/src/main/java/org/embulk/EmbulkEmbed.java]: It seems the import is not used (since past version).

Comment (dmikurube) [reply embulk-core/src/main/java/org/embulk/EmbulkEmbed.java]: Ah, good catch! Thanks.

File: embulk-core/src/main/java/org/embulk/EmbulkEmbed.java [hunk 2]
```diff
@@ -1,165 +1,240 @@
...
-        public Bootstrap addModules(Iterable<? extends Module> additionalModules) {
-            final List<Module> copy = ImmutableList.copyOf(additionalModules);
-            return overrideModules(new Function<List<Module>, Iterable<Module>>() {
+        public Bootstrap addModules(final Iterable<? extends Module> additionalModules) {
+            final ArrayList<Module> copyMutable = new ArrayList<>();
+            for (final Module module : additionalModules) {
+                copyMutable.add(module);
+            }
+            final List<Module> copy = Collections.unmodifiableList(copyMutable);
+            return overrideModules(new com.google.common.base.Function<List<Module>, Iterable<Module>>() {
```
[the remark this answers is not in the document]
Comment (dmikurube) [reply embulk-core/src/main/java/org/embulk/EmbulkEmbed.java]: Good catch! I mechanically replaced Guava Function to Java 8 Function, but that'd be better.

**Comment by dmikurube** [comment]
Thanks! Merging this with 22e5b754188df0ccfd83341f1a647e7f509f895b after the CI gets happy.

**Pull request merged**